### DISCUSSION TITLE
How to use `wry` API in a Tauri app?

### DISCUSSION THREAD

**Discussion by MikeChongCan** [tauri-apps/tauri#7528] [General]
Background: I'm thinking about using `wry` as a headless web view renderer for a Tauri app.

However, I can open a empty web view like this but cannot do anything more.

It feels like the new window is blocked.

I would also like to use a web view derived from Tauri runtime if possible. But I'm not sure how to get access to wry WebView APIs?

```rs
 fn fetch(
        &self,
        url: String,
        callback: Box<
            dyn Fn(Result<String, String>) + Send + Sync,
        >,
    ) {
       // running in a non-main thread
        use wry::{
            application::{
                event::{Event, StartCause, WindowEvent},
                event_loop::{ControlFlow, EventLoop},
                window::WindowBuilder,
            },
            webview::WebViewBuilder,
        };
        let sender = irondash_run_loop::RunLoop::sender_for_main_thread();

        match sender {
            Ok(handle) => {
                handle.send(move || {
                    let event_loop = EventLoop::new();
                    let window = WindowBuilder::new()
                        .with_title("Hello World")
                        .with_closable(true)
                        .build(&event_loop)
                        .unwrap();

                    let webview = WebViewBuilder::new(window)
                        .unwrap()
                        // .with_url(url.as_str())
                        // .unwrap()
                        .with_devtools(true)
                        .build()
                        .unwrap();

                    webview.load_url(url.as_str());
                    // webview.open_devtools();

                    println!("url: -> {:}", webview.url());

                    println!("opening url: {}", url);
                });
            }
            Err(e) => {
                println!("e: {:?}", e);
            }
        }
    }

```

I can create a Tauri window with external URLs.
However, by default, Tauri disable ipc between this external web page and my Rust code.

I hope to have a safe way to communicate between external web and Rust

**Comment by FabianLars** [comment]
> How to use wry API in a Tauri app?

You can't reallyyy do that (in a sane way on all platforms).

> However, by default, Tauri disable ipc between this external web page and my Rust code.

Did you know about https://tauri.app/v1/api/config/#securityconfig.dangerousremotedomainipcaccess already? It's a config to allow ipc access for external urls.

**Reply by MikeChongCan** [reply]
Thank you!

I don't know whether we can use it for all external URLs because my use case is kind of special, which is to load every possible websites / URLs

**Reply by JVariance** [reply]
> Thank you!
> 
> I don't know whether we can use it for all external URLs because my use case is kind of special, which is to load every possible websites / URLs

Have you found a solution?

**Comment by oddpxl** [comment]
@imWildCat Did you get this to work ? *curious*

**Reply by MikeChongCan** [reply]
no I use electron for now
